### DISCUSSION TITLE
Training model with more images

### DISCUSSION THREAD

**Discussion by Codemagnett** [ultralytics/ultralytics#8677] [Q&A]
I have recently started to use yolov8 model, I trained model with 400 images further I want to train 200 images what will be the process?

**Comment by glenn-jocher** [comment]
Hello! Great to hear you're using YOLOv8. 😊 To continue training your model with an additional 200 images, you can simply add these images to your dataset, ensuring they're properly annotated. Then, resume training using the last saved checkpoint. Here's a quick example on how to do this:

```bash
yolo detect train data=your_dataset.yaml model=path/to/last_saved_model.pt epochs=additional_epochs
```

Make sure `your_dataset.yaml` includes paths to both your original 400 images and the new 200 images. This way, your model will continue learning from where it left off, now with the added data. Happy training!

**Reply by Codemagnett** [reply]
Thank you for the reply, can you clarify whether the retraining will be done on the all 600 images or only on newly added 200 images and I don't have your_dataset.yaml file I have config.yaml file .I am using the below given code snipet for training a model can you please tell me the what are the changes to be done in the code snipet for training 200 new images

import os
from ultralytics import YOLO

model = YOLO("yolov8l.yaml")
results = model.train(data=os.path.join(ROOT_DIR, "config.yaml"), epochs=50)

**Reply by glenn-jocher** [reply]
@Codemagnett hello! 😊 When you continue training with the additional 200 images, it's best to train on all 600 images (the original 400 plus the new 200). This approach ensures your model learns from the full dataset, improving its accuracy and generalization.

Regarding your `config.yaml` file, it's perfectly fine to use it under a different name than `your_dataset.yaml`. The key is to ensure it points to all 600 images for training. If your `config.yaml` already does that, you're good to go!

For training with the new images, your code snippet looks almost ready. Just make sure your `config.yaml` includes paths to all 600 images. If it's already set up for the 400 images, you'll need to update it to include the new 200 images. Here's a quick reminder of how your code looks and is ready for training with the updated dataset:

```python
import os
from ultralytics import YOLO

# Assuming ROOT_DIR is correctly defined and includes the updated config.yaml
model = YOLO("yolov8l.yaml")
results = model.train(data=os.path.join(ROOT_DIR, "config.yaml"), epochs=50)
```

That's it! Your model will now train on the entire dataset of 600 images. Happy training, and feel free to reach out if you have more questions! 🚀
